### ISSUE TITLE
Event Loop context in the debugger

### ISSUE THREAD

**Issue by puzpuzpuz** [nodejs/help#1621]
Hi guys.

I wonder if there is a way to see current event loop phase and current microtasks queues (and other task queues) during debugging a node application? I haven't found such stuff on the net, but may be it's possible.

It may be useful in certain cases like the example that is described in the recent [blog post](https://v8.dev/blog/fast-async) from V8's team:
```js
const p = Promise.resolve();

(async () => {
  await p;
  console.log('after:await');
})();

p.then(() => console.log('tick:a'))
  .then(() => console.log('tick:b'));
```

I know that this particular example is quite synthetic. But there are quite many real-world scenarios when being able to put a breakpoint somewhere in your code and see the context of the event loop (and other related stuff like the worker thread pool) execution would be very helpful.

So I wonder if it's somehow available in the node itself or any 3rd-party libraries at the moment?

**Comment by keenondrums** [comment]
+1. Could be very useful for debugging and educative purposes

**Comment by devsnek** [comment]
node itself doesn't have access to the event loop. this request should go to the relevant engines's debugging tools (v8 and chakra debuggers)

**Comment by gberdzenishvili** [comment]
+1 it will allow to get better understanding of how event loop works

**Comment by keenondrums** [comment]
@devsnek couldn't it be at core level  

https://github.com/nodejs/node/blob/v10.x-staging/deps/uv/src/unix/core.c#L350

I thought we could just expose somehow which function inside uv_run is currently running

**Comment by devsnek** [comment]
@keenondrums the js engine keeps which things are in the event loop, libuv just tells it when to handle those things

**Comment by keenondrums** [comment]
@devsnek wouldn't it mean that we could track which phase of the loop is currently running at libuv level ad address a specific engine only for contents of the queues?

**Comment by devsnek** [comment]
@keenondrums yeah, but I think it would make sense for engines just to integrate this into their existing debug tooling. the contents of the microtasks queue isn't even exposed to node, we just tell it to run.

**Comment by puzpuzpuz** [comment]
Posted an issue in V8's bug tracker: https://bugs.chromium.org/p/v8/issues/detail?id=8479
